sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -710,4 +710,11 @@ class JsonFunctionsSuite extends QueryTest with SharedSparkSession {
       Seq(Row("string")))
   }
 
+  test("json_array_length") {
+    val df = Seq(1).toDF("json")
+    val errMsg = intercept[AnalysisException] {
+      df.selectExpr("json_array_length(json)")
+    }.getMessage
+    assert(errMsg.contains("due to data type mismatch"))
+  }
```

Comment:
Shall we remove this because this is already covered at `json-functions.sql` more extensively.